### PR TITLE
Python: Bump to python 3.8

### PR DESCRIPTION
**Pull request by Spredzy**
##### SUMMARY

Bumping default python used in awx and ansible venv to python 3.8

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API
 

##### AWX VERSION

 - devel

##### ADDITIONAL INFORMATION

 - N/A

### CODE REVIEW

**Comment by Spredzy** [comment]
Pausing work on this PR. Work will resume when the execution enviironment work will land. Removing the need for an ansible venv (and only having an awx one)

**Comment by Spredzy** [comment]
recheck

**Comment by unlikelyzero** [comment]
@Spredzy hey, this may impact our ui tests' use of awxkit. Let us know with @ui-qe when you're about 2 days away from merging this and we'll work to get it ready on our side

**Comment by unlikelyzero** [comment]
@Spredzy , give ui-qe some time to fix our pipelines? I think @one-t is the most knowledgeable about how to test the changes.

@one-t do you mind taking this one?

**Comment by ryanpetrello** [comment]
@Spredzy is this PR still active (it's out of date given the k8s installer -> operator move)?  Would it make sense to tackle this later, or do you think we'll update this PR in the near future?

**Comment by unlikelyzero** [comment]
Adding @tiagodread

File: Makefile [hunk 1]
```diff
@@ -299,8 +299,8 @@ swagger: reports
 check: flake8 pep8 # pyflakes pylint
 
 awx-link:
-	[ -d "/awx_devel/awx.egg-info" ] || python3 /awx_devel/setup.py egg_info_dev
-	cp -f /tmp/awx.egg-link /var/lib/awx/venv/awx/lib/python$(PYTHON_VERSION)/site-packages/awx.egg-link
+	[ -d "/awx_devel/awx.egg-info" ] || $(PYTHON) /awx_devel/setup.py egg_info_dev
+	cp -f /tmp/awx.egg-link /venv/awx/lib/python$(PYTHON_VERSION)/site-packages/awx.egg-link
```
Comment (ryanpetrello) [inline Makefile]: This looks out of date (`/venv/awx/`)?

**Review by ryanpetrello — COMMENTED** [review]
Once Zuul is happy, one minor comment, otherwise, this looks about right to me 👍

**Comment by unlikelyzero** [comment]
@Spredzy has this gone through the API and CLI tests?

**Review by ryanpetrello — APPROVED** [review]

**Comment by Spredzy** [comment]
recheck

**Comment by Spredzy** [comment]
recheck

File: requirements/requirements.txt [hunk 2]
```diff
@@ -83,6 +83,8 @@ defusedxml==0.6.0
     #   social-auth-core
 dictdiffer==0.8.1
     # via openshift
+distro==1.5.0
+    # via -r /awx_devel/requirements/requirements.in
```
Comment (wenottingham) [inline requirements/requirements.txt:87]: You can/should regenerate this with 3.8. Some things will fall out (and you can remove the license files for them)

Or that can be a followup change.

Comment (Spredzy) [reply requirements/requirements.txt:87]: Will do in a subsequent PR yep

**Comment by Spredzy** [comment]
recheck

**Pull request merged**